tests/Backend/Mixed/Workspaces/S3WorkspacesTest.php

```diff
@@ -615,59 +349,200 @@ public function testLoadIncrementalNotNullable($backend, $bucketBackend)
+
+        $options = [
+            "input" => [
+                [
+                    "source" => "in.c-mixed-test-{$bucketBackend}.dates",
+                    "destination" => "dates",
+                    "columns" => [
+                        [
+                            'source' => 'valid_from',
+                            'type' => 'TIMESTAMP',
+                        ],
+                    ],
+                ],
+            ],
         ];
+
+        // exception should not be thrown, date conversion should be applied
+        $workspaces->loadWorkspaceData($workspace['id'], $options);
+
+        $wsBackend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+
+        $data = $wsBackend->fetchAll("dates", \PDO::FETCH_ASSOC);
+        $this->assertCount(3, $data);
     }
 
-    public function workspaceMixedAndSameBackendDataWithDataTypes()
+    /**
+     * @dataProvider workspaceMixedBackendData
+     * @param $backend
+     */
+    public function testMixedBackendWorkspaceLoad($backend, $bucketBackend)
     {
-        $simpleDataTypesDefinitionSnowflake = [
-            [
-                'source' => 'price',
-                'type' => 'VARCHAR',
-            ],
+        if ($this->_client->bucketExists("out.c-mixed-test-" . $bucketBackend)) {
+            $this->_client->dropBucket(
+                "out.c-mixed-test-{$bucketBackend}",
+                [
+                    'force' => true,
+                ]
+            );
+        }
+
+        if ($this->_client->bucketExists("in.c-mixed-test-" . $bucketBackend)) {
+            $this->_client->dropBucket(
+                "in.c-mixed-test-{$bucketBackend}",
+                [
+                    'force' => true,
+                ]
+            );
+        }
+
+        $bucketId = $this->_client->createBucket("mixed-test-{$bucketBackend}", "in", "", $bucketBackend);
+        $outBucketId =  $this->_client->createBucket("mixed-test-{$bucketBackend}", "out", "", $bucketBackend);
+
+        //setup test table
+        $table1Id = $this->_client->createTable(
+            $bucketId,
+            'languages',
+            new CsvFile(__DIR__ . '/../../../_data/languages.csv'),
             [
-                'source' => 'quantity',
-                'type' =>'NUMBER',
-            ],
-        ];
-        $simpleDataTypesDefinitionRedshift = [
+                'primaryKey' => 'id',
+            ]
+        );
+
+        $table2Id = $this->_client->createAliasTable(
+            $outBucketId,
+            $table1Id,
+            'Languages'
+        );
+
+        $table3Id = $this->_client->createAliasTable(
+            $outBucketId,
+            $table1Id,
+            'LanguagesOneColumn',
             [
-                'source' => 'price',
-                'type' => 'VARCHAR',
-            ],
+                'aliasColumns' => [
+                    'id',
+                ],
+            ]
+        );
+
+        $table4Id = $this->_client->createAliasTable(
```

Comment:
Proč se u S3 testují takhle komplikované scénáře a u ABS ne?

Reply:
(teda ne že bych chápal, proč je takhle komplikovaný scénář potřeba, když jde o to jestli se to správně vyexportuje do S3/ABS)

Reply:
Tady sem právě pochopil, že ten test je v projektu mezi Snowflake-Redshift a Redshift nemáme podporu pro ABS nevím teda jestli ju vůbec má a určitě ju nepotřebujeme.

Reply:
Jo, jasně. Už to chápu... S3 se testuje napříč backendy ;) OK